sqlglot/dialects/hive.py

```diff
@@ -562,13 +562,18 @@ def datatype_sql(self, expression: exp.DataType) -> str:
                 expression = exp.DataType.build("text")
             elif expression.this in exp.DataType.TEMPORAL_TYPES:
                 expression = exp.DataType.build(expression.this)
-            elif expression.is_type("float"):
-                size_expression = expression.find(exp.DataTypeParam)
-                if size_expression:
+
+            size_expression = expression.find(exp.DataTypeParam)
+            if size_expression:
+                if expression.is_type("float"):
                     size = int(size_expression.name)
                     expression = (
                         exp.DataType.build("float") if size <= 32 else exp.DataType.build("double")
                     )
+                elif size_expression.expression:
+                    # This removes keywords like `BYTE` (Oracle) in `CAST(x AS VARCHAR(5 BYTE))`
+                    expression = expression.copy()
+                    t.cast(exp.DataTypeParam, expression.find(exp.DataTypeParam)).expression.pop()
```

Comment:
On second thought, maybe we should control this using a generator flag instead so that we only generate these specifiers for Oracle?

Reply:
sure

Reply:
Done